File: 350-intersection-of-two-arrays-ii/intersection-of-two-arrays-ii.java
class Solution {
    public int[] intersect(int[] nums1, int[] nums2) {

        HashMap<Integer, Integer> map1 = new HashMap<>();
        HashMap<Integer, Integer> map2 = new HashMap<>();
        ArrayList<Integer> ans = new ArrayList<>();

        for (int ele : nums1){
            if(map1.containsKey(ele)){
                int freq = map1.get(ele);
                map1.put(ele, freq + 1);
            }
            else{
                map1.put(ele, 1);
            }
        }

        for (int ele : nums2){
            if(map2.containsKey(ele)){
                int freq = map2.get(ele);
                map2.put(ele, freq + 1);
            }
            else{
                map2.put(ele, 1);
            }
        }


        for (int key : map1.keySet()){
            if(map2.containsKey(key)){
                int freq1 = map1.get(key);
                int freq2 = map2.get(key);

                int min = Math.min(freq1, freq2);

                for(int i=0; i<min; i++)ans.add(key); 
            }
        }
            
        return ans.stream()
                .mapToInt(Integer::intValue)
                .toArray();

    }
}

File: 350-intersection-of-two-arrays-ii/intersection-of-two-arrays-ii.py
class Solution(object):
    def intersect(self, nums1, nums2):
        
        map1 = {}
        map2 = {}
        ans = []

        for ele in nums1:
            if ele in map1:
                map1[ele] += 1
            else:
                map1[ele] = 1 
        
        for ele in nums2:
            if ele in map2:
                map2[ele] += 1
            else:
                map2[ele] = 1 
        
        for key in map1.keys():
            if key in map2.keys():
                n = min(map1[key], map2[key])
                for i in range(n):
                    ans.append(key)

        return ans